### ISSUE TITLE
Commented CSV rows with quotes throw exceptions at the start of the next read line.

### ISSUE THREAD

**Issue by asgardaesir** [JoshClose/CsvHelper#561]
Hey there,

I recently switched to CSVHelper and while quickly testing some expected scenarios I found that having a row which starts with a comment containing a quote character will cause the next line to be rejected.

In the following example Country4 is commented out and an exception occurs on reading the file. The BadDataCallback string says it is the line with at 'Country5' the beginning.

```
Code,Name
Country1,Country 1
Country2,Country 2
Country3,"Country""Extra quotes Excel Scenario""3"
#Country4,"""Country Extra quotes SQL export Scenario 4"""
Country5,XBHD??
Country101,\\xyz//defqqq
Country10,Czech Republic/Czechia
Country11,Italy\Italia
```

If the quotes are removed from the commented line, the load succeeds.

Here is the nunit test I ran to replicate this issue:

```
[Test]
        public void CsvCommentError()
        {
            var streamReader = new StreamReader(@"C:\temp\testcsv.csv");

            ICsvParser parser = new CsvParser(streamReader, new CsvConfiguration
            {
                Delimiter = ",",
                Comment = '#',
                BufferSize = 1024,
                Quote = '"',
                AllowComments = true,
                HasHeaderRecord = false,
                BadDataCallback = s =>
                {
                    throw new Exception(string.Format("Bad CSV data detected while reading file"));
                }
            });

            var _csvReader = new CsvHelper.CsvReader(parser);

            while (_csvReader.Read())
            {
                
            }
        }
```

Thanks for making and maintaining this library.

Brad.

**Comment by JoshClose** [comment]
This seems to be fixed in 3.0. I ran it in 2.x and got `Country5` showing up as bad. I ran it in 3.0 and nothing shows up.

3.0 is in beta on NuGet right now. I will hopefully be releasing it soon, but I want to get through as many tickets as I can before I do that. I want to make sure any more breaking changes happen before I release, otherwise they'll have to wait for 4.0.

**Comment by JoshClose** [comment]
Btw, here is the test code I used.

```cs
void Main()
{
	var config = new CsvConfiguration
	{
		Delimiter = ",",
		Comment = '#',
		BufferSize = 1024,
		Quote = '"',
		AllowComments = true,
		HasHeaderRecord = false,
		BadDataCallback = s => $"Bad data: {s}".Dump()
	};
	using (var stream = new MemoryStream())
	using (var writer = new StreamWriter(stream))
	using (var reader = new StreamReader(stream))
	using (var csv = new CsvReader(reader, config))
	{		
		writer.WriteLine("Code,Name");
		writer.WriteLine("Country1, Country 1");
		writer.WriteLine("Country2,Country 2");
		writer.WriteLine("Country3,\"Country\"\"Extra quotes Excel Scenario\"\"3\"");
		writer.WriteLine("#Country4,\"\"\"Country Extra quotes SQL export Scenario 4\"\"\"");
		writer.WriteLine("Country5,XBHD ??");
		writer.WriteLine("Country101,\\xyz//defqqq");
		writer.WriteLine("Country10, Czech Republic / Czechia");
		writer.WriteLine("Country11,Italy\\Italia");
		writer.Flush();
		stream.Position = 0;

		while (csv.Read())
		{
		}
	}
}
```
